crates/red_knot_python_semantic/resources/mdtest/boundness_declaredness.md

```diff
@@ -0,0 +1,202 @@
+# Boundness and declaredness
+
+This document demonstrates how type-inference and diagnostics works for public symbols. We test the
```

Comment:
Maybe clarify scoping of this document. Here at top-level we mention "public symbols", but then there's a heading `## Public symbols` below, and the name of the file doesn't mention "public", so it seems a little unclear whether this document is intended to (eventually?) cover boundedness and declaredness for non-public cases as well.

(Personally I would be inclined to say this document is large enough already, and well-scoped as-is, so I'd probably clarify the headings/names to explicitly make this whole document about public symbols. But either way is fine.)